source/Personality.cpp

```diff
@@ -59,7 +59,10 @@ namespace {
 		LAUNCHING,
 		DARING,
 		SECRETIVE,
-		RAMMING
+		RAMMING,
+
+		// This must be last so it can be used for bounds checking.
+		LAST_ITEM_IN_PERSONALITY_TRAIT_ENUM
```

Comment:
I would call it something less of a mouthful, like PERSONALITY_COUNT or TRAIT_COUNT for example.

Reply:
I chose a long name intentionally, so it would be impossible to miss. It's less a variable name, and more a reminder of a big bug you'll introduce if you mess up the ordering.

Reply:
The variable the rest of the code should access is `Personality::PERSONALITY_COUNT`, so you won't see `LAST_ITEM_IN_PERSONALITY_TRAIT_ENUM` anywhere, except the one place it is being compared to `PERSONALITY_COUNT` in the static_assert.